example_dags/example_dynamic_task_template.py

```diff
@@ -0,0 +1,43 @@
+import os
+from datetime import datetime
+
+from airflow import DAG
+from airflow.decorators import task
+
+from astro import sql as aql
+from astro.files import get_file_list
+from astro.files.base import File
+from astro.sql.table import Metadata, Table
+
+GCS_BUCKET = os.getenv("GCS_BUCKET", "gs://dag-authoring")
+ASTRO_GCP_CONN_ID = os.getenv("ASTRO_GCP_CONN_ID", "google_cloud_default")
+ASTRO_BIGQUERY_DATASET = os.getenv("ASTRO_BIGQUERY_DATASET", "dag_authoring")
+
+
+# [START howto_operator_get_file_list]
+@task
+def load_to_bigquery(path):
+    aql.load_file(
```

Comment:
Would this also work?
```
load_to_bigquery = aql.load_file(...)
```

Reply:
ok, I ``path``  param to initialize ``File`` object. How would I pass that?

Reply:
check https://github.com/astronomer/astro-sdk/pull/596#discussion_r943745295

Reply:
```
load_to_bigquery = aql.load_file(
    input_file=File(path=GCS_BUCKET),
    output_table=Table(
        metadata=Metadata(
            schema=ASTRO_BIGQUERY_DATASET,
        ),
        conn_id=ASTRO_GCP_CONN_ID,
    ),
    use_native_support=False,
)
```

Reply:
but @kaxil GCS_BUCKET path is dynamic here, isn't it?

Reply:
Aaah yes, I get it now, we probably need docstring to explain that

Reply:
ok

Reply:
ok after looking at this closely, there are 2 issue:

1) If you have the following, you are running a task within a task, I don't think it will return anything.
```
@task
def load_to_bigquery(path):
    aql.load_file(
```

2) The `load_file` function or `LoadFile` operator expects a `File` object whereas `get_file_list` returns a "list of string". We have 2 options: 1) Make `get_file_list` return a list of File objects and 2) Make `LoadFile`/`load_file` accept a "string" along with the `File` objects so users can use it with non-Astro SDK tasks and Dynamic Task Mapping.

Reply:
We should make the following example work to address the larger issue mentioned in the last comment:

```
with DAG(
    "example_dynamic_task_template",
    schedule_interval=None,
    start_date=datetime(2000, 1, 1),
    catchup=False,
) as dag:

    imdb_movies = LoadFile.partial(
        task_id="load_csv",
        output_table=Table(name="imdb_movies_2", conn_id="sqlite_default"),
    ).expand(input_file=get_file_list(path=GCS_BUCKET, conn_id=ASTRO_GCP_CONN_ID))
```

Reply:
@kaxil to your first point, I think get_file_list should absolutely return a list of File objects. Each of the tasks created need that context to properly process the file. Files should be the defacto way to pass files between tasks, anything else would be off.

To your second point, maybe there is a larger discussion about adding dynamic task mapping functionality to xcomargs? I think that makes sense if people expect to treat them like operator objects.

Reply:
re: 1) Yup updated the PR with `get_file_list` returning list of `File` objects.

re: 2) Since `XComArg` already instantiates an operator, we can't have `partial` and `expand` as some of the params/args are required and will raise error. For example, `load_file` function already instantiates `LoadFileOperator` so we "can't not" pass `input_file` or else it will raise a required param not passed error. We can only do that on "class level" so not sure if it is feasible.